wgpu/src/svg.rs

```diff
@@ -0,0 +1,604 @@
+                    },
+                    write_mask: wgpu::ColorWrite::ALL,
+                }],
+                depth_stencil_state: None,
+                index_format: wgpu::IndexFormat::Uint16,
+                vertex_buffers: &[
+                    wgpu::VertexBufferDescriptor {
+                        stride: mem::size_of::<Vertex>() as u64,
+                        step_mode: wgpu::InputStepMode::Vertex,
+                        attributes: &[wgpu::VertexAttributeDescriptor {
+                            shader_location: 0,
+                            format: wgpu::VertexFormat::Float2,
+                            offset: 0,
+                        }],
+                    },
+                    wgpu::VertexBufferDescriptor {
+                        stride: mem::size_of::<Instance>() as u64,
+                        step_mode: wgpu::InputStepMode::Instance,
+                        attributes: &[
+                            wgpu::VertexAttributeDescriptor {
+                                shader_location: 1,
+                                format: wgpu::VertexFormat::Float2,
+                                offset: 0,
+                            },
+                            wgpu::VertexAttributeDescriptor {
+                                shader_location: 2,
+                                format: wgpu::VertexFormat::Float2,
+                                offset: 4 * 2,
+                            },
+                        ],
+                    },
+                ],
+                sample_count: 1,
+                sample_mask: !0,
+                alpha_to_coverage_enabled: false,
+            });
+
+        let vertices = device
+            .create_buffer_mapped(QUAD_VERTS.len(), wgpu::BufferUsage::VERTEX)
+            .fill_from_slice(&QUAD_VERTS);
+
+        let indices = device
+            .create_buffer_mapped(QUAD_INDICES.len(), wgpu::BufferUsage::INDEX)
+            .fill_from_slice(&QUAD_INDICES);
+
+        let instances = device.create_buffer(&wgpu::BufferDescriptor {
+            size: mem::size_of::<Instance>() as u64,
+            usage: wgpu::BufferUsage::VERTEX | wgpu::BufferUsage::COPY_DST,
+        });
+
+        Pipeline {
+            cache: RefCell::new(Cache::new()),
+
+            pipeline,
+            uniforms: uniforms_buffer,
+            vertices,
+            indices,
+            instances,
+            constants: constant_bind_group,
+            texture_layout,
+        }
+    }
+
+    fn load(&self, handle: &Handle) {
+        if !self.cache.borrow().contains(&handle) {
+            if !handle.path.is_file() {
+                let mem = Memory::NotFound;
+
+                let _ = self.cache.borrow_mut().insert(&handle, mem);
+            }
+
+            let mut opt = resvg::Options::default();
+            opt.usvg.path = Some(handle.path.clone());
+            opt.usvg.dpi = handle.dpi as f64;
+            opt.usvg.font_size = handle.font_size as f64;
```

Comment:
Could we use `DrawTarget::set_transform` to set the DPI scaling in `upload`? Everything should scale nicely and we will avoid the weird `96.0` and `20.0` magic constants.

Reply:
Those constants are for usvg which takes them to create the node tree. I'm not entirely sure how they affect the image creation but I think they influence how `<text>` tags in the svg file are processed.

Reply:
The idea is to leave everything in default settings and simply apply a uniform scaling to everything.

Reply:
Usvg does not use the dpi setting to create bigger images. It's just for font rendering inside the icon.

Reply:
Right, I don't think we should alter that setting. Applying a transformation to the `DrawTarget` should be enough.

Reply:
Sure, I can change it to keep the standard settings. I think this is a corner case if one at all anyway.